docs/design/sustainability/README.md

```diff
@@ -1,75 +1,57 @@
+| | | Use serverless functions | [Energy Proportionality](./sustainable-engineering-principles.md#energy-proportionality) | [Serverless Architecture Design](https://docs.microsoft.com/en-us/azure/architecture/serverless-quest/serverless-overview) | |
+| | For services with co-located users (EG internal employee apps) | Select a datacenter region that is physicaly close to them | [Networking](./sustainable-engineering-principles.md#networking) | [Azure products available by region](https://azure.microsoft.com/en-us/global-infrastructure/services/) | |
+| | | Consider running an edge device | [Networking](./sustainable-engineering-principles.md#networking) | [Azure Stack Edge](https://azure.microsoft.com/en-us/products/azure-stack/edge/) | [Understanding edge tradeoffs](./sustainable-action-disclaimers.md#action-consider-running-an-edge-device) |
+| | For services with global users | Cache data close to them using a [CDN](https://en.wikipedia.org/wiki/Content_delivery_network) | [Networking](./sustainable-engineering-principles.md#networking) | [Benefits of a CDN](https://docs.microsoft.com/en-us/azure/frontdoor/front-door-overview#key-benefits) | |
+| | For systems sending data over the network | Send only the data that will be used | [Networking](./sustainable-engineering-principles.md#networking) | | |
+| | | Compress data to reduce the size | [Networking](./sustainable-engineering-principles.md#networking) | [Compressing and extracting files in .NET](https://docs.microsoft.com/en-us/dotnet/standard/io/how-to-compress-and-extract-files) | |
+| | When designing for the end user | Consider giving users visibility and control over their energy usage | [Electricity Consumption](./sustainable-engineering-principles.md#electricity-consumption) [Demand Shaping](./sustainable-engineering-principles.md#demand-shaping) | [Designing for eco-mode](https://principles.green/principles/demand-shaping/#heading-eco-modes) | |
+| | | Design and test your application to be compatible for a wide variety of devices, especially older devices | [Embodied Carbon](./sustainable-engineering-principles.md#embodied-carbon) | [Extending device lifespan](https://medium.com/google-design/to-make-apps-accessible-make-them-compatible-with-different-devices-11298c6d3f06) [Compatibility Testing](https://www.techdim.com/what-is-compatibility-testing/) | |
+| | When selecting a programming language | Consider the energy efficiency of languages | [Electricity Consumption](./sustainable-engineering-principles.md#electricity-consumption) | [Reasoning about the energy consumption of programming languages](https://thenewstack.io/which-programming-languages-use-the-least-electricity/), [Programming Language Energy Efficiency [PDF]](https://greenlab.di.uminho.pt/wp-content/uploads/2017/10/sleFinal.pdf) | [Making informed progamming language choices](./sustainable-action-disclaimers.md#action-consider-the-energy-efficiency-of-languages) |
+
```

Comment:
```suggestion
| | Time-shifting workloads | Consider postponing the workload to a time where the carbon intensity (g CO2/kWh) is lower | [Electricity Consumption](./sustainable-engineering-principles.md#electricity-consumption) | [Green Software Foundation Carbon Aware SDK](https://github.com/Green-Software-Foundation/carbon-aware-sdk) | [Making use of green electricity](./sustainable-action-disclaimers.md#demand-shaping) |
| | Location-shifting workloads | Consider moving the workload to another location/Azure Region where the carbon intensity (g CO2/kWh) is lower | [Electricity Consumption](./sustainable-engineering-principles.md#electricity-consumption) | [Green Software Foundation Carbon Aware SDK](https://github.com/Green-Software-Foundation/carbon-aware-sdk) | [Making use of green electricity](./sustainable-action-disclaimers.md#demand-shaping) |

```

Reply:
These are on our list, but did not make the cut for this first round of checklist usage because the recommendation currently requires too much subject matter expertise to be actionable. But I would expect this down the road.

Reply:
I'd b e the first to drive this!... but agree for now we need more guidance before people go crazy with carbon aware compute.  Possibly something to do alongside the Sustainability Culture of Learning content so there is more guidance (videos etc).